controller/app.controller.js

```diff
@@ -7,9 +12,24 @@ const getTopics = (req, res, next) => {
 };
 
 const getAll = (req, res, next) => {
-  readAll().then((endpoints) => {
-    res.status(200).send(endpoints);
+  return Promise.resolve(endpoints).then((result) => {
+    res.status(200).send({ endpoints: result });
   });
 };
 
-module.exports = { getTopics, getAll };
+const getArticleById = (req, res, next) => {
+  const { article_id } = req.params;
+  selectArticleById(article_id)
+    .then((article) => {
+      if (!article) {
+        return Promise.reject({
```

Comment:
rejecting could be done in the model, it would make your controller slightly neater, the controller usually just handles the flow of the request and response :)